analytics/src/analytics_ios.mm

```diff
@@ -373,6 +373,44 @@ void SetSessionTimeoutDuration(int64_t milliseconds) {
       setSessionTimeoutInterval:static_cast<NSTimeInterval>(milliseconds) / kMillisecondsPerSecond];
 }
 
+void SetDefaultEventParameters(const std::map<std::string, firebase::Variant>& default_parameters) {
+  FIREBASE_ASSERT_RETURN_VOID(internal::IsInitialized());
+  NSMutableDictionary* ns_default_parameters =
+      [[NSMutableDictionary alloc] initWithCapacity:default_parameters.size()];
+  for (const auto& pair : default_parameters) {
+    NSString* key = SafeString(pair.first.c_str());
+    const Variant& value = pair.second;
+
+    if (value.is_null()) {
+      [ns_default_parameters setObject:[NSNull null] forKey:key];
+    } else if (value.is_int64()) {
+      [ns_default_parameters setObject:[NSNumber numberWithLongLong:value.int64_value()]
+                                forKey:key];
+    } else if (value.is_double()) {
+      [ns_default_parameters setObject:[NSNumber numberWithDouble:value.double_value()] forKey:key];
+    } else if (value.is_string()) {
+      [ns_default_parameters setObject:SafeString(value.string_value()) forKey:key];
+    } else if (value.is_bool()) {
+      [ns_default_parameters setObject:[NSNumber numberWithBool:value.bool_value()] forKey:key];
+    } else if (value.is_vector() || value.is_map()) {
+      LogError("SetDefaultEventParameters: Value for key '%s' has type '%s' which is not supported "
+               "for default event parameters. Only string, int64, double, bool, and null are "
+               "supported. Skipping.",
+               pair.first.c_str(), Variant::TypeName(value.type()));
+    } else {
+      LogError("SetDefaultEventParameters: Value for key '%s' has an unexpected type '%s' which is "
+               "not supported. Skipping.",
+               pair.first.c_str(), Variant::TypeName(value.type()));
+    }
+  }
+  [FIRAnalytics setDefaultEventParameters:ns_default_parameters];
```

Comment:
Double-check FIRAnalytics setDefaultEventParameters -- if we pass in an empty dictionary, will it clear the parameters? This could happen if every single value we specified was an invalid type.